File: Desktop/dsa/Search and Sort/selectionSort.py

#fHandle = open("input.txt", 'r')
#A = [int(num) for num in (fHandle.read()).split()]

def selectionSort(A):
	for startPoint in range(len(A)-1):
		
		searchMin = startPoint
		min = A[searchMin]
		indexMin = searchMin
		
		for index in range(serachmin+1, len(A)):
			if A[index] < min:
				min = A[index]
				indexMin = index
		
		
		A[indexMin] = A[startPoint]
		A[startPoint] = min
	
	return A
	
A = [3, 2, 6, 2, 6, 5, 3, 8]
B = selectionSort(A)
print B

File: Desktop/dsa/Search and Sort/selectionSort.cpp
#include<iostream>
using namespace std;

void selectionSort(int A, const int n){


	for(int i = 0; i < n; i++){
	int index = i, min = A[i], temp;

		for(int j = i; j<n; j++){
			if(A[j] < min){
				index = j;
				min = A[j];
			}			
		}

	temp = A[index];
	A[index] = A[i];
	A[i] = temp;
	}

}

int main(){

	int A[] = {2, 4, 1, 65, 7, 4, 3, 23};

	selectionSort(A, 8);
	
	for(int i = 0; i < 8; i++){

		cout<<A[i]<<" ";
	}
	cout<<endl;
	return 0;
}
